internal/cmd/io.go

```diff
@@ -36,3 +39,17 @@ type UpstreamIO interface {
 	// return `""`.
 	Terminal() bool
 }
+
+func NewUpstreamIO(ctx context.Context, id string, stdout string, stderr string, stdin string, terminal bool) (UpstreamIO, error) {
```

Comment:
might help to add a comment that when we use a binary logger stdout and stderr should be the same path?